startup.sh

```diff
@@ -1,172 +1,621 @@
-    *) echo "Wrong option. Try again";drivessd;;
-esac
+ssd_drive() {
+    title "SSD Drive Confirmation"
+    read -r -p "Is this system using an SSD? [like yes/no]: " _SSD
+    case "$_SSD" in
+    y | Y | yes | Yes | YES)
+        set_option "SSD" 1
+        set_option "MOUNTOPTION" "noatime,compress=zstd,ssd,commit=120"
+        ;;
+    n | N | no | NO | No)
+        set_option "SSD" 0
+        set_option "MOUNTOPTION" "noatime,compress=zstd,commit=120"
+        ;;
+    *)
+        invalid_option
+        ssd_drive
+        ;;
+    esac
 }
 
-# selection for disk type
-diskpart () {
-# show disks present on system
-lsblk -n --output TYPE,KNAME,SIZE | awk '$1=="disk"{print NR,"/dev/"$2" - "$3}' # show disks with /dev/ prefix and size
-echo -ne "
-------------------------------------------------------------------------
-    THIS WILL FORMAT AND DELETE ALL DATA ON THE DISK             
-    Please make sure you know what you are doing because         
-    after formating your disk there is no way to get data back      
-------------------------------------------------------------------------
+disk_selection() {
+    title "Disk Selection"
+    DISKLIST="$(lsblk -n --output TYPE,KNAME,SIZE | awk '$1=="disk"{print "/dev/"$2" - "$3}')" # show disks with /dev/ prefix and size
+    PS3="$PROMPT"
+    select _DISK in "${DISKLIST[@]}"; do
+        if elements_present "$_DISK" "${DISKLIST[@]}"; then
+            DISK=$(echo "$_DISK" | awk '{print $1}')
+            set_option "DISK" "$DISK"
+            break
+        else
+            invalid_option
+            disk_selection
+            break
+        fi
+    done
+}
 
-Please enter full path to disk: (example /dev/sda):
-"
-read option
-echo "DISK=$option" >> setup.conf
+user_info() {
+    title "Add Your Information"
+    while true; do
+        read -r -p "Please enter your username [default is archtitus]: " USERNAME
+        if [[ -z "$USERNAME" ]]; then
+            set_option "USERNAME" "archtitus"
+            break
+        elif [[ "${USERNAME,,}" =~ ^[a-z_]([a-z0-9_-]{0,31}|[a-z0-9_-]{0,30}\$)$ ]]; then
+            set_option "USERNAME" "${USERNAME,,}" # convert to lower case as in issue #109
+            break
+        else
+            invalid_option
+            continue
+        fi
+    done
+    set_password "PASSWORD"
+    while true; do
+        read -r -p "Please enter your hostname [default is archlinux]: " HOSTNAME
+        if [[ -z "$HOSTNAME" ]]; then
+            set_option "HOSTNAME" "archlinux"
+            break
+        elif [[ "${HOSTNAME,,}" =~ ^[a-z][a-z0-9_.-]{0,62}[a-z0-9]$ ]]; then
+            set_option "HOSTNAME" "${HOSTNAME,,}"
+            break
+        else
+            invalid_option
+            continue
+        fi
+    done
+}
 
-drivessd
-set_option DISK $option
+set_locale() {
+    title "Setup Locale"
+    LOCALES=($(grep UTF-8 /etc/locale.gen | sed 's/\..*$//' | sed '/@/d' | awk '{print $1}' | uniq | sed 's/#//g'))
```

Comment:
I like the intention of not missing out on locales to show the user... but it's a list of 303 locales and I can't even see the one I intend to select on screen since they all dont fit on screen. 
![image](https://user-images.githubusercontent.com/1778670/154892284-1f858930-a228-47b0-ac15-9006b25376bf.png)